Client/Frontend/Browser/TabLocationView.swift

```diff
@@ -430,4 +434,28 @@ private class DisplayTextField: UITextField {
     fileprivate override var canBecomeFirstResponder: Bool {
         return false
     }
+    
+    override func textRect(forBounds bounds: CGRect) -> CGRect {
+        var rect: CGRect = super.textRect(forBounds: bounds)
+        
+        if let size: CGSize = (self.text as NSString?)?.size(withAttributes: [.font: self.font!]) {
+            if size.width > self.bounds.width {
+                rect.origin.x = rect.origin.x - (size.width - self.bounds.width)
+                rect.size.width = rect.size.width + (size.width - self.bounds.width)
+            }
+        }
+        return rect
+    }
+}
+
+extension String {
+    func etldValue(parser: DomainParser?) -> String {
+        guard parser != nil else {
+            return self
+        }
+        if let url: URL = URL(string: self.trimmingCharacters(in: CharacterSet(charactersIn: "/"))), let host: String = url.host {
```

Comment:
I would like to see some unit tests for this method, with few success/failure branches

Reply:
Added some test cases, Most were already there in the library. To be noted punycode is not handled by lib